lib/api/objectCopy.js

```diff
@@ -91,12 +93,20 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
         });
         return { error: userMetadata };
     }
-    // If metadataDirective is 'COPY' and location constraint header is
-    // specified, new location constraint should override source object's
-    if (whichMetadata === 'COPY' && headers[locationHeader]) {
-        userMetadata[locationHeader] = headers[locationHeader];
+    // If metadataDirective is:
+    // - 'COPY' and source object has a locaction contrain in its metadata
+    // we use the bucket destination location constraint
```

Comment:
also, just for clarification, this new functionality is:
with MD directive 'COPY', the new object will be put to the destination bucket location, even if the source object has a location constraint in its metadata?